### DISCUSSION TITLE
Berry & OLED

### DISCUSSION THREAD

**Discussion by martin072** [arendst/Tasmota#19525] [Q&A] [answered]
Does Berry has the option to code directly to an attached OLED display? Does this also work with LVGL (although with limited support I assume)? Or do you need to issue tasmota Display commands from Berry?

Thanks!

**Comment by s-hadinger** [comment]
As long as the display uses a supported driving chip, it will work. What hardware do you have in mind?

**Reply by martin072** [reply]
It’s a standard cheap Ali 128*64 black and white i2c oled screen. Standard supported by tasmota I’ve noticed, but just wondered if there is anything special for Berry. I.e use the standard display commands, or is there a way to code it directly in Berry?

**Reply by s-hadinger** [reply]
For Berry there is LVGL and HaspMOTA

**Reply by martin072** [reply]
I’ve noticed, does LVGL also work with a very simple OLED screen? seems it’s more for color and touch  displays?

**Reply by s-hadinger** [reply]
I'm confused what you call OLED - which is just an alternative technology to LCD screen.

Do I assume you are talking about monochrome display? (OLED can be in color)

**Reply by martin072** [reply]
Sorry. Yes. Monochrome 128x64

**Reply by s-hadinger** [reply]
Yes, LVGL supports monochrome. There is a special theme for that, although it's been a while so I don't remember everything.

There are also pixel perfect monochrome fonts for those small screens.

**Reply by martin072** [reply]
Ok thanks. Will dive into it

**Reply by sfromis** [reply]
Is the display SSD1306 or what?

**Reply by martin072** [reply]
Yes

**Reply by sfromis** [reply, marked as answer]
https://github.com/arendst/Tasmota/blob/development/tasmota/displaydesc/SSD1306_128x64_display.ini

**Reply by martin072** [reply]
Thanks!
